### DISCUSSION TITLE
Qual a melhor forma de organizar os controllers e keys

### DISCUSSION THREAD

**Discussion by DIMAAGR** [Flutterando/forum#29] [Perguntas e Respostas] [answered]
Olá a todos, estou com um problema, na verdade uma dúvida sobre a forma de organizar o código, bom, sobre controllers como o TextEdittingController e keys como o ScaffoldKey, como devo organizá-los no código, devo colocá-los nas stores ou deixá-los na tela, ou até mesmo criar uma classe a parte para fazer essa organização?

**Como eu faço hoje:** bom, para falar a verdade atualmente faço da seguinte forma: deixo as keys na store, inicializando-as dentro da tela, e movendo-os para as stores, os controllers atualmente deixo dentro de uma classe como por exemplo um RegisterControllers, onde todos os controllers de todos os campos do registro estão lá.

**Por que a pergunta:** Depois de refatorar o projeto pela segunda vez, os desenvolvedores sêniors não gostaram da forma de organização do projeto, e uma das coisas que eles resolveram modificar foi isso, quero entender bem sobre como organizar melhor o projeto para não ter que passar pelo mesmo novamente!

**Comment by sergiotucano** [comment, marked as answer]
Não existe resposta certa ou forma correta.
Se o textController dizer sentido só no escopo daquele widget não tem problema criá-lo e destruí-lo ali mesmo.

Porém se o Controller tem uso fora daquele widget é interessante deixar ele numa classe de controle, principalmente se for utilizá-lo com algum gerenciador de estado.

No seu caso é melhor conversar com os devs que não gostaram da forma que tu fez e perguntar o porquê não gostaram e pedir perguntar o que eles fariam e o porquê fariam daquela forma

**Reply by DIMAAGR** [reply]
Vlw cara pela ajuda!!!
